projects/components/src/select/select.component.ts

```diff
@@ -34,18 +34,41 @@ import { SelectSize } from './select-size';
         this.groupPosition,
         selected ? selected.style.toString() : '',
         this.showBorder ? 'border' : '',
-        this.disabled ? 'disabled' : ''
+        this.disabled ? 'disabled' : '',
+        this.triggerDisplayMode
       ]"
       *htLetAsync="this.selected$ as selected"
     >
-      <ht-popover [disabled]="this.disabled" [closeOnClick]="true" class="select-container">
+      <ht-popover
+        [disabled]="this.disabled"
+        [closeOnClick]="true"
+        class="select-container"
+        [ngSwitch]="this.triggerDisplayMode"
+      >
         <ht-popover-trigger>
-          <div class="trigger-content" [ngClass]="this.justifyClass">
-            <ht-icon *ngIf="this.icon" class="trigger-prefix-icon" [icon]="this.icon" size="${IconSize.Small}">
-            </ht-icon>
+          <div
+            *ngSwitchCase="'${SelectTriggerDisplayMode.MenuWithBorder}'"
+            class="trigger-content"
+            [ngClass]="this.justifyClass"
+          >
+            <ht-icon *ngIf="this.icon" class="trigger-prefix-icon" [icon]="this.icon" [size]="this.iconSize"> </ht-icon>
             <ht-label class="trigger-label" [label]="selected?.label || this.placeholder"> </ht-label>
             <ht-icon class="trigger-icon" icon="${IconType.ChevronDown}" size="${IconSize.ExtraSmall}"> </ht-icon>
           </div>
+          <div
+            *ngSwitchCase="'${SelectTriggerDisplayMode.Icon}'"
+            class="trigger-content icon-only"
+            [ngClass]="this.selected !== undefined ? 'selected' : ''"
+          >
+            <ht-icon
+              class="icon"
+              *ngIf="this.icon"
+              [icon]="this.icon"
+              [size]="this.iconSize"
+              [htTooltip]="this.placeholder"
```

Comment:
This should be the selected label if selected, right? matching line 55 - could move that logic together if you want.